tests/unit/core/api/test_index.js

```diff
@@ -41,7 +41,28 @@ describe(__filename, () => {
         .then(() => mockWindow.verify());
     });
 
-    it('encodes non-ascii URLs in UTF8', () => {
+    it('does not encode non-ascii URLs in UTF8 on the client', () => {
+      const endpoint = 'project-ă-ă-â-â-日本語';
+      mockWindow.expects('fetch')
+        .withArgs(`${apiHost}/api/v3/${endpoint}/`, {
+          body: undefined, credentials: undefined, method: 'GET', headers: {},
+        })
+        .once()
+        .returns(createApiResponse());
+
+      // We use `cloneDeep()` to allow modifications on the `config` object,
+      // since a call to `get()` makes it immutable.
+      const clientConfig = configUtil.cloneDeep(config);
+      clientConfig.client = true;
+      clientConfig.server = false;
+
+      return api.callApi({
```

Comment:
Does this test fail without your patch? I can't think of how it would. I think you won't really be able to make it fail since our tests run in NodeJS, not the browser. Thus, I think you should just make an assertion about not encoding the query string in the client. This isn't the greatest test but it at least solidifies the *intention* of the code. You can use a stub for that:

```js
const fakeUtf8Encode = sinon.stub();

return api.callApi({
  _config: clientConfig,
  _utf8: { encode: fakeUtf8Encode },
  endpoint,
})
  .then(() => mockWindow.verify())
  .then(() => {
    sinon.assert.notCalled(fakeUtf8Encode);
  });
```

Reply:
I'd say you do not need to do the inverse for the server-behavior test because in that case you *will* be able to reproduce the failure.

Reply:
Well it fails actually:

```
  ● /Users/william/projects/mozilla/addons-frontend/tests/unit/core/api/test_index.js › core.callApi › does not encode non-ascii URLs in UTF8 on the client

    ExpectationError: Unexpected call: fetch(https://localhost/api/v3/project-Ä-Ä-Ã¢-Ã¢-æ¥æ¬èª/, { body: undefined, credentials: undefined, headers: {  }, method: "GET" })
        Expected fetch(https://localhost/api/v3/project-ă-ă-â-â-日本語/, { body: undefined, credentials: undefined, headers: {  }, method: "GET" }[, ...]) once (never called)
```

Reply:
ok, duh. I see. It fails without your patch because the old code was doing the UTF8 encoding. Alright, this test is good. There is no need to introduce a new stub for the utf8 encoder.